docs/apache-airflow-providers-google/connections/gcp_sql.rst

```diff
@@ -76,3 +76,37 @@ Extra (optional)
     .. code-block:: bash
 
         export AIRFLOW_CONN_GOOGLE_CLOUD_SQL_DEFAULT='gcpcloudsql://user:XXXXXXXXX@1.1.1.1:3306/mydb?database_type=mysql&project_id=example-project&location=europe-west1&instance=testinstance&use_proxy=True&sql_proxy_use_tcp=False'
+
+Configuring and using IAM authentication
+----------------------------------------
+
+.. warning::
+  IAM authentication working only for Google Service Accounts.
+
+Configure Service Accounts on Google Cloud IAM side
+"""""""""""""""""""""""""""""""""""""""""""""""""""
+
+For connecting via IAM you need to use Service Account. It can be the same service account which you use for
+the ``gcloud`` authentication or an another account. If you decide to use a different account then this
+account should be impersonated from the account which used for ``gcloud`` authentication and granted
+a ``Service Account Token Creator`` role. More information how to grant a role `here
+<https://cloud.google.com/iam/docs/manage-access-service-accounts?hl=en&_gl=1*3bsv5i*_ga*NDY4NDIyNTcxLjE3MjkxNzQ4MTM.*_ga_WH2QY8WWF5*MTcyOTE5MzU1OS4yLjEuMTcyOTE5NTM0My4wLjAuMA..#single-role>`_.
+
+Also the Service Account should be configured for working with IAM.
+Here are links describing what should be done before the start:
+`PostgreSQL<https://cloud.google.com/sql/docs/postgres/iam-logins#before_you_begin>`_ and
+`MySQL<https://cloud.google.com/sql/docs/mysql/iam-logins#before_you_begin>`_.
+
+Configure ``gcpcloudsql`` connection with IAM enabling
+""""""""""""""""""""""""""""""""""""""""""""""""""""""
+
+For using IAM you need to enable ``"use_iam": "True"`` in the ``extra`` field. And specify IAM account in this format
```

Comment:
If this works only for Postgres and MySQL we should mention it here in docs.

Reply:
In the beginer of this documentation it is already mentioned that this connection works only for two types of databases [here](https://github.com/apache/airflow/blob/main/docs/apache-airflow-providers-google/connections/gcp_sql.rst?plain=1#L25C1-L27C1).  Also Connection and Operators for CloudSQL work only with two databases MySQL and Postgres. It was hard coded in the Hook and the code has a [function](https://github.com/apache/airflow/blob/main/providers/src/airflow/providers/google/cloud/hooks/cloud_sql.py#L958) for checking database's type.

Reply:
ok ,sg.